backend/src/plugins/addons/server/services/seed/backup/download.ts

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs';
+import http from 'http';
+import https from 'https';
+import path from 'path';
+import { getBackupPath } from './export';
+
+export interface DownloadBackup {
+   strapiOrigin: string;
+}
+
+export async function downloadBackup({
+   strapiOrigin,
+}: DownloadBackup): Promise<void> {
+   const exportURL = new URL(
+      getBackupPath({ isEncrypted: false }),
+      strapiOrigin
+   );
+   try {
+      const backupFilePath = getBackupFilePath({ isEncrypted: false });
+      ensureDirectoryExistence(backupFilePath);
+      strapi.log.info(`🌱 Downloading backup from: ${exportURL.href}`);
+      await downloadFile(exportURL, backupFilePath);
+   } catch (err) {
+      strapi.log.error(
+         `🌱💥 Failed to download backup from: ${exportURL.href}`
+      );
+      console.error(err);
+      throw new Error(`Failed to download backup from: ${exportURL.href}`);
+   }
+}
+
+interface GetImportPathInput {
+   isEncrypted: boolean;
+}
+
+export function getBackupFilePath({ isEncrypted }: GetImportPathInput) {
```

Comment:
This seems like a duplicate function from `import.ts`. Since `seed.ts` looks like it expects the downloaded backup and import path to be the same, perhaps `downloadBackup()` could return a file path, and `importBackup()` could accept a file path as a param/option.

Reply:
Great suggestion! 👏